Sources/NIOHTTP1/HTTPPipelineSetup.swift

```diff
@@ -389,69 +389,63 @@ extension ChannelPipeline {
-    ///   - pipelining: Whether to provide assistance handling HTTP clients that pipeline
-    ///         their requests. Defaults to `true`. If `false`, users will need to handle
-    ///         clients that pipeline themselves.
-    ///   - upgrade: Whether to add a `HTTPServerUpgradeHandler` to the pipeline, configured for
-    ///         HTTP upgrade. Defaults to `nil`, which will not add the handler to the pipeline. If
-    ///         provided should be a tuple of an array of `HTTPServerProtocolUpgrader` and the upgrade
-    ///         completion handler. See the documentation on `HTTPServerUpgradeHandler` for more
-    ///         details.
-    ///   - errorHandling: Whether to provide assistance handling protocol errors (e.g.
-    ///         failure to parse the HTTP request) by sending 400 errors. Defaults to `true`.
-    ///   - headerValidation: Whether to validate outbound request headers to confirm that they meet
-    ///         spec compliance. Defaults to `true`.
-    ///   - encoderConfiguration: The configuration for the ``HTTPResponseEncoder``.
-    /// - Return: An `EventLoopFuture` that will fire when the pipeline is configured. The future contains an `EventLoopFuture`
-    /// 
+    ///   - isPipeliningEnabled: Whether to provide assistance handling HTTP clients that pipeline
+    ///   their requests. Defaults to `true`. If `false`, users will need to handle clients that pipeline themselves.
+    ///   - isErrorHandlingEnabled: Whether to provide assistance handling protocol errors (e.g. failure to parse the HTTP
+    ///   request) by sending 400 errors. Defaults to `true`.
+    ///   - isResponseHeaderValidationEnabled: Whether to validate outbound response headers to confirm that they are
+    ///   spec compliant. Defaults to `true`.
+    ///   - httpResponseEncoderConfiguration: The configuration for the ``HTTPResponseEncoder``.
+    ///   - httpServerUpgradeConfiguration: The configuration for the ``NIOTypedHTTPServerUpgradeHandler``.
+    /// - Returns: An `EventLoopFuture` that will fire when the pipeline is configured. The future contains an `EventLoopFuture`
+    /// that is fired once the pipeline has been upgraded or not and contains the `UpgradeResult`.
     @available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
     @_spi(AsyncChannel)
     public func configureTypedHTTPServerPipeline<UpgradeResult: Sendable>(
-        withPipeliningAssistance pipelining: Bool = true,
-        withServerUpgrade upgrade: NIOTypedHTTPServerUpgradeConfiguration<UpgradeResult>,
-        withErrorHandling errorHandling: Bool = true,
-        withOutboundHeaderValidation headerValidation: Bool = true,
-        withEncoderConfiguration encoderConfiguration: HTTPResponseEncoder.Configuration = .init()
+        isPipeliningEnabled: Bool = true,
+        isErrorHandlingEnabled: Bool = true,
+        isResponseHeaderValidationEnabled: Bool = true,
```

Comment:
I think `is` isn't right here, because it reads like a question. I think `enablePipelining`, `enableErrorHandling`, and `enableResponseHeaderValidation` are more appropriate